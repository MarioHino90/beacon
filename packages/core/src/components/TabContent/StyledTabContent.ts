/**
 * @fileoverview the styled tab content container
 */
import classNames from "classnames";
import styled from "styled-components";
import { getSpacing } from "@braze/beacon-styling";

import { StyledTabContentProps } from "./types";

export const StyledTabContent = styled.div.attrs(({ className }) => ({
  className: classNames(className, "bcl-tab-content"),
}))<StyledTabContentProps>`
  margin: ${getSpacing("xl")} 0;
`;
