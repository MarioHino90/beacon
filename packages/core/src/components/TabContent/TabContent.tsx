/**
 * @fileoverview component that handles showing/hiding tab content - as well as
container styling for tab content
 */
import React, { ReactElement } from "react";

import { TabContentProps } from "./types";
import { StyledTabContent } from "./StyledTabContent";

export const TabContent = ({
  active,
  ...props
}: TabContentProps): ReactElement | null =>
  active ? <StyledTabContent {...props} /> : null;
