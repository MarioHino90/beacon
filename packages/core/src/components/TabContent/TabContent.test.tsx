import React from "react";
import { shallow } from "enzyme";

import { TabContent } from "./TabContent";

describe("TabContent", () => {
  it("should render correctly", () => {
    const wrapper = shallow(<TabContent active />);

    expect(wrapper).toMatchSnapshot();
  });

  it("should render correctly if not active", () => {
    const wrapper = shallow(<TabContent active={false} />);

    expect(wrapper).toMatchSnapshot();
  });

  it("should render correctly with className", () => {
    const wrapper = shallow(<TabContent active className="someClass" />);

    expect(wrapper).toMatchSnapshot();
  });
});
