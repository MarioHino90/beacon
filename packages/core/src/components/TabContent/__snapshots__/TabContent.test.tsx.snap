// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TabContent should render correctly 1`] = `<StyledTabContent />`;

exports[`TabContent should render correctly if not active 1`] = `""`;

exports[`TabContent should render correctly with className 1`] = `
<StyledTabContent
  className="someClass"
/>
`;
